### ISSUE TITLE
GUI/Control node in Viewport tree does not preview in layout mode in editor 

### ISSUE THREAD

**Issue by goatchurchprime** [godotengine/godot#39387]
**Godot version:**
3.2.1.stable

**OS/device including version:**
Windows10

**Issue description:**
The main scene of the demo, `gui_in_3d.tscn`, includes the scene `gui_panel_3d.tscn` which contains a Viewport and a 2D GUI Control CanvasItem node.  This 2D GUI Control node contains a panel with some buttons, sliders and other GUI components.  

When you run the demo, the code in `gui_3d.gd` maps mouse click events on the 3D rendered image to mouse click events on the 2D GUI Control, and it all works fine.

But, when you select the GUI in the node tree to make changes to the layout, all you see is a blank grey rectangle in the editor.

There's also a second problem -- if you merge the nodes from the scene into the main scene (to avoid the complexity of having a subscene), then the GUI panel appears as a black rectangle in 3D with no texture.  The mouse clicks still work, though.

It is a common requirement for little text based notices to appear in 3D above an object, for example to identify the object's name or to briefly flash up the points score for having shot it.  The use of a Viewport is currently the [only way](https://godotengine.org/qa/45599/flat-text-in-a-3d-environment) for producing this feature.  It would be nice if there was something like a `TextTexture` that could get it done in a more light-weight manner.

**Steps to reproduce:**
Load the [gui_in_3d](https://github.com/godotengine/godot-demo-projects/tree/master/viewport/gui_in_3d) demo project 

You can preview the `gui_panel_3d.tscn` scene in in 3D, like so:

![image](https://user-images.githubusercontent.com/677254/84015003-1f51a900-a973-11ea-9682-4f2d4ea6c939.png)

But when you select the GUI Control node, the preview is blank:

![image](https://user-images.githubusercontent.com/677254/84017111-1d3d1980-a976-11ea-9b5e-926fd4e03857.png)

One work-around is to temporarily drag the GUI Control node out of the Viewport node, make changes, and then drag it back into the Viewport node, but this shouldn't be necessary.

![image](https://user-images.githubusercontent.com/677254/84017203-3e056f00-a976-11ea-8fdb-bf006b351db3.png)

**Minimal reproduction project:**
Problem is produced directly on the demo project here:
https://github.com/godotengine/godot-demo-projects/tree/master/viewport/gui_in_3d

**Comment by Calinou** [comment]
I'm fairly sure there's an open issue about this already, but I can't find it right now.

**Comment by Zireael07** [comment]
It was mentioned several times, yes, but I'm not sure we have a **dedicated** issue about it.

**Comment by KoBeWi** [comment]
I don't think this is a bug. The GUI is in another Viewport from editor's and with this scene setup it has nowhere to render to.

**Comment by Zireael07** [comment]
I am not sure what to classify it as, but it's a real annoyance for working with 2d in 3d and the reverse...

**Comment by Tooniis** [comment]
Moving the control node out of the viewport temporarily also means you don't get the boundaries of that viewport in the editor, which can make framing and sizing elements more difficult, although a possible workaround is to use guides.

**Comment by goatchurchprime** [comment]
I think this is also tied to https://github.com/godotengine/godot/issues/11727

(It seemed impossible for me to find this issue using the github search engine, and I was reduced to going back through my contribution history to look for it.)

I just wanted to report a workflow work-around I'm using, which is to make a Control node as a sibling to the Viewport called "ViewportFake" and then put the following code in the `_ready()` function

```GDScript
	var fgui = $ViewportFake.get_node_or_null("GUI")
	if fgui != null:
		$ViewportFake.remove_child(fgui)
		$Viewport.add_child(fgui)
```
Now I can happily iterate on my GUI design and keep rerunning the game without too much hassle.  Unfortunately if there is more than one GUI panel it will show up in the same 2D preview because there is only one.
